Build problems with Maven 3.9.0 and missing BasicRepositoryConnectorFactory

In the last two weeks Github Actions has been updated to use Maven 3.9.0 and now previously working jobs using the gluonfx-maven-plugin fail with
```
Error:  Failed to execute goal com.gluonhq:gluonfx-maven-plugin:1.0.16:compile (default-cli) on project commlink6: Execution default-cli of goal com.gluonhq:gluonfx-maven-plugin:1.0.16:compile failed: A required class was missing while executing com.gluonhq:gluonfx-maven-plugin:1.0.16:compile: org/eclipse/aether/connector/basic/BasicRepositoryConnectorFactory
Error:  -----------------------------------------------------
Error:  realm =    plugin>com.gluonhq:gluonfx-maven-plugin:1.0.16
```
It seems to be related to this discussion [here](https://www.mail-archive.com/dev%40maven.apache.org/msg128736.html)

Older versions of Maven still work fine.

i have the same problem while trying to compile and create windows executable.
`[ERROR] Failed to execute goal com.gluonhq:gluonfx-maven-plugin:1.0.16:nativerun (default-cli) on project densimetro: Execution default-cli of goal com.gluonhq:gluonfx-maven-plugin:1.0.16:nativerun failed: A required class was missing while executing com.gluonhq:gluonfx-maven-plugin:1.0.16:nativerun: org/eclipse/aether/connector/basic/BasicRepositoryConnectorFactory
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>com.gluonhq:gluonfx-maven-plugin:1.0.16
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/C:/Users/Informatica/.m2/repository/com/gluonhq/gluonfx-maven-plugin/1.0.16/gluonfx-maven-plugin-1.0.16.jar
[ERROR] urls[1] = file:/C:/Users/Informatica/.m2/repository/com/gluonhq/substrate/0.0.56/substrate-0.0.56.jar
[ERROR] urls[2] = file:/C:/Users/Informatica/.m2/repository/com/googlecode/plist/dd-plist/1.23/dd-plist-1.23.jar
[ERROR] urls[3] = file:/C:/Users/Informatica/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.49/bcpkix-jdk15on-1.49.jar
[ERROR] urls[4] = file:/C:/Users/Informatica/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.49/bcprov-jdk15on-1.49.jar
[ERROR] urls[5] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/svm/21.1.0/svm-21.1.0.jar
[ERROR] urls[6] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-linux-amd64/21.1.0/svm-hosted-native-linux-amd64-21.1.0.tar.gz
[ERROR] urls[7] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-darwin-amd64/21.1.0/svm-hosted-native-darwin-amd64-21.1.0.tar.gz
[ERROR] urls[8] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-windows-amd64/21.1.0/svm-hosted-native-windows-amd64-21.1.0.tar.gz
[ERROR] urls[9] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/sdk/graal-sdk/21.1.0/graal-sdk-21.1.0.jar
[ERROR] urls[10] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/objectfile/21.1.0/objectfile-21.1.0.jar
[ERROR] urls[11] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/nativeimage/pointsto/21.1.0/pointsto-21.1.0.jar
[ERROR] urls[12] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/compiler/compiler/21.1.0/compiler-21.1.0.jar
[ERROR] urls[13] = file:/C:/Users/Informatica/.m2/repository/org/graalvm/truffle/truffle-api/21.1.0/truffle-api-21.1.0.jar
[ERROR] urls[14] = file:/C:/Users/Informatica/.m2/repository/org/apidesign/bck2brwsr/aot/0.53/aot-0.53.jar
[ERROR] urls[15] = file:/C:/Users/Informatica/.m2/repository/org/apidesign/bck2brwsr/vm4brwsr/0.53/vm4brwsr-0.53.jar
[ERROR] urls[16] = file:/C:/Users/Informatica/.m2/repository/org/apidesign/bck2brwsr/core/0.53/core-0.53.jar
[ERROR] urls[17] = file:/C:/Users/Informatica/.m2/repository/com/google/javascript/closure-compiler/v20210601/closure-compiler-v20210601.jar
[ERROR] urls[18] = file:/C:/Users/Informatica/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[19] = file:/C:/Users/Informatica/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[ERROR] urls[20] = file:/C:/Users/Informatica/.m2/repository/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.jar
[ERROR] urls[21] = file:/C:/Users/Informatica/.m2/repository/org/codehaus/plexus/plexus-java/0.9.11/plexus-java-0.9.11.jar
[ERROR] urls[22] = file:/C:/Users/Informatica/.m2/repository/org/ow2/asm/asm/7.0-beta/asm-7.0-beta.jar
[ERROR] urls[23] = file:/C:/Users/Informatica/.m2/repository/com/thoughtworks/qdox/qdox/2.0-M9/qdox-2.0-M9.jar
[ERROR] urls[24] = file:/C:/Users/Informatica/.m2/repository/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar
[ERROR] urls[25] = file:/C:/Users/Informatica/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.25/plexus-interpolation-1.25.jar
[ERROR] urls[26] = file:/C:/Users/Informatica/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.4/plexus-sec-dispatcher-1.4.jar
[ERROR] urls[27] = file:/C:/Users/Informatica/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[28] = file:/C:/Users/Informatica/.m2/repository/org/apache/maven/maven-builder-support/3.6.0/maven-builder-support-3.6.0.jar
[ERROR] urls[29] = file:/C:/Users/Informatica/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.2.1/maven-shared-utils-3.2.1.jar
[ERROR] urls[30] = file:/C:/Users/Informatica/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[ERROR] urls[31] = file:/C:/Users/Informatica/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.3/org.eclipse.sisu.inject-0.3.3.jar
[ERROR] urls[32] = file:/C:/Users/Informatica/.m2/repository/com/google/inject/guice/4.2.1/guice-4.2.1-no_aop.jar
[ERROR] urls[33] = file:/C:/Users/Informatica/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[ERROR] urls[34] = file:/C:/Users/Informatica/.m2/repository/com/google/guava/guava/25.1-android/guava-25.1-android.jar
[ERROR] urls[35] = file:/C:/Users/Informatica/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
[ERROR] urls[36] = file:/C:/Users/Informatica/.m2/repository/org/checkerframework/checker-compat-qual/2.0.0/checker-compat-qual-2.0.0.jar
[ERROR] urls[37] = file:/C:/Users/Informatica/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar
[ERROR] urls[38] = file:/C:/Users/Informatica/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar
[ERROR] urls[39] = file:/C:/Users/Informatica/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar
[ERROR] urls[40] = file:/C:/Users/Informatica/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar
[ERROR] urls[41] = file:/C:/Users/Informatica/.m2/repository/org/apache/commons/commons-lang3/3.8.1/commons-lang3-3.8.1.jar
[ERROR] urls[42] = file:/C:/Users/Informatica/.m2/repository/org/apache/maven/shared/maven-invoker/2.2/maven-invoker-2.2.jar
[ERROR] urls[43] = file:/C:/Users/Informatica/.m2/repository/org/eclipse/aether/aether-transport-http/1.1.0/aether-transport-http-1.1.0.jar
[ERROR] urls[44] = file:/C:/Users/Informatica/.m2/repository/org/apache/httpcomponents/httpclient/4.3.5/httpclient-4.3.5.jar
[ERROR] urls[45] = file:/C:/Users/Informatica/.m2/repository/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
[ERROR] urls[46] = file:/C:/Users/Informatica/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar
[ERROR] urls[47] = file:/C:/Users/Informatica/.m2/repository/org/slf4j/jcl-over-slf4j/1.6.2/jcl-over-slf4j-1.6.2.jar
[ERROR] urls[48] = file:/C:/Users/Informatica/.m2/repository/org/eclipse/aether/aether-transport-file/1.1.0/aether-transport-file-1.1.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR]
[ERROR] -----------------------------------------------------
[ERROR] : org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException`

--- a/pom.xml
+++ b/pom.xml
@@ -1,5 +1,5 @@
 <!--
- ~ Copyright (c) 2019, 2021, Gluon
+ ~ Copyright (c) 2019, 2023, Gluon
  ~ All rights reserved.
  ~
  ~ Redistribution and use in source and binary forms, with or without
@@ -34,7 +34,7 @@
   <groupId>com.gluonhq</groupId>
   <artifactId>gluonfx-maven-plugin</artifactId>
   <packaging>maven-plugin</packaging>
-  <version>1.0.17-SNAPSHOT</version>
+  <version>1.0.20-SNAPSHOT</version>
   <name>GluonFX plugin for Maven</name>
   <description>GluonFX plugin allows to run JavaFX application on the JVM or to create their native images.</description>
   <inceptionYear>2019</inceptionYear>
@@ -44,9 +44,11 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
+    <source.plugin.version>3.3.0</source.plugin.version>
     <javadoc.plugin.version>3.5.0</javadoc.plugin.version>
-    <gpg.plugin.version>3.0.1</gpg.plugin.version>
-    <substrate.version>0.0.57-SNAPSHOT</substrate.version>
+    <gpg.plugin.version>3.1.0</gpg.plugin.version>
+    <maven.resolver.version>1.9.15</maven.resolver.version>
+    <substrate.version>0.0.60-SNAPSHOT</substrate.version>
   </properties>
 
   <dependencies>
@@ -58,13 +60,13 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
-      <version>3.9.0</version>
+      <version>3.9.4</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.plugin-tools</groupId>
       <artifactId>maven-plugin-annotations</artifactId>
-      <version>3.8.1</version>
+      <version>3.9.0</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
@@ -80,7 +82,7 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
-      <version>3.9.0</version>
+      <version>3.9.4</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
@@ -98,17 +100,17 @@
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-transport-http</artifactId>
-      <version>1.9.6</version>
+      <version>${maven.resolver.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-transport-file</artifactId>
-      <version>1.9.6</version>
+      <version>${maven.resolver.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-connector-basic</artifactId>
-      <version>1.9.6</version>
+      <version>${maven.resolver.version}</version>
     </dependency>
   </dependencies>
 
@@ -132,12 +134,12 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.9.0</version>
+        <version>3.11.0</version>
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-plugin-plugin</artifactId>
-        <version>3.8.1</version>
+        <version>3.9.0</version>
       </plugin>
       <plugin>
         <groupId>org.sonatype.plugins</groupId>
@@ -153,7 +155,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
-        <version>3.2.1</version>
+        <version>3.4.1</version>
         <executions>
           <execution>
             <id>enforce-no-snapshots</id>
@@ -184,7 +186,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-source-plugin</artifactId>
-            <version>3.2.1</version>
+            <version>${source.plugin.version}</version>
             <executions>
               <execution>
                 <id>attach-sources</id>

--- a/pom.xml
+++ b/pom.xml
@@ -44,8 +44,8 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
-    <javadoc.plugin.version>3.1.0</javadoc.plugin.version>
-    <gpg.plugin.version>1.6</gpg.plugin.version>
+    <javadoc.plugin.version>3.5.0</javadoc.plugin.version>
+    <gpg.plugin.version>3.0.1</gpg.plugin.version>
     <substrate.version>0.0.57-SNAPSHOT</substrate.version>
   </properties>
 
@@ -58,12 +58,13 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
-      <version>3.6.0</version>
+      <version>3.9.0</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.plugin-tools</groupId>
       <artifactId>maven-plugin-annotations</artifactId>
-      <version>3.6.0</version>
+      <version>3.8.1</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
@@ -79,33 +80,35 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
-      <version>3.6.0</version>
+      <version>3.9.0</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-invoker</artifactId>
-      <version>2.2</version>
+      <version>3.2.0</version>
     </dependency>
 
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-aether-provider</artifactId>
       <version>3.3.9</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.aether</groupId>
-      <artifactId>aether-transport-http</artifactId>
-      <version>1.1.0</version>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-transport-http</artifactId>
+      <version>1.9.6</version>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.aether</groupId>
-      <artifactId>aether-transport-file</artifactId>
-      <version>1.1.0</version>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-transport-file</artifactId>
+      <version>1.9.6</version>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.aether</groupId>
-      <artifactId>aether-connector-basic</artifactId>
-      <version>1.1.0</version>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-connector-basic</artifactId>
+      <version>1.9.6</version>
     </dependency>
   </dependencies>
 
@@ -129,17 +132,17 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.8.1</version>
+        <version>3.9.0</version>
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-plugin-plugin</artifactId>
-        <version>3.6.0</version>
+        <version>3.8.1</version>
       </plugin>
       <plugin>
         <groupId>org.sonatype.plugins</groupId>
         <artifactId>nexus-staging-maven-plugin</artifactId>
-        <version>1.6.7</version>
+        <version>1.6.13</version>
         <extensions>true</extensions>
         <configuration>
           <serverId>ossrh</serverId>
@@ -150,7 +153,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
-        <version>3.0.0-M3</version>
+        <version>3.2.1</version>
         <executions>
           <execution>
             <id>enforce-no-snapshots</id>
@@ -181,7 +184,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-source-plugin</artifactId>
-            <version>2.2.1</version>
+            <version>3.2.1</version>
             <executions>
               <execution>
                 <id>attach-sources</id>

